### PR TITLE
Added ability to set connectTimeout of socket

### PR DESCRIPTION
**Pull request by MattyBoy4444**
This allows you to change the timeout of the Socket.  When a socket dies, this is the timeout that is used to determine if it should reconnect.

### CODE REVIEW

**Comment by mraleph** [comment]
Thanks for the contribution. This PR does not pass analysis unfortunately. See errors above.

**Comment by MattyBoy4444** [comment]
Fixed

**Comment by mraleph** [comment]
Thanks! One more request: could you add a `CHANGELOG.md` entry describing the change

```
## 3.1.1-dev

* Added ...
```

and change `pubspec.yaml` to set the version `3.1.1-dev` as well. 

After that I will merge.

**Comment by MattyBoy4444** [comment]
Done. Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@
   `:authority` value for UDS connections to `localhost` instead of using
   UDS path. Using path triggers checks in HTTP2 servers which
   attempt to validate `:authority` value.
+* `ChannelOptions` now exposes `connectTimeout`, which is used on the
```
Comment (mraleph) [inline CHANGELOG.md:10]: This is in the wrong section. It needs to go into `## 3.1.1-dev`.

Comment (MattyBoy4444) [reply CHANGELOG.md]: Done

**Pull request merged**